src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1557,6 +1569,17 @@ protected virtual void ApplyCompilationOptionsChanged(ProjectId projectId, Compi
             this.OnCompilationOptionsChanged(projectId, options);
         }
 
+        /// <summary>
+        /// This method is called during <see cref="TryApplyChanges(Solution)"/> to change the default namespace.
+        ///
+        /// Override this method to implement the capability of changing default namespace.
+        /// </summary>
+        protected virtual void ApplyDefaultNamespaceChanged(ProjectId projectId, string? defaultNamespace)
```

Comment:
Not sure if this should be the right API here or whether we just want this to be ApplyProjectInfoChanged, because there's other things that aren't being considered here, and I'm not sure I want to create a new method for each thing...

Reply:
I think it makes sense as is, because there are already public methods per property (`WithDefaultNamespace()`, `WithHasAllInformation()` and so on) so it would be symmetrical to have apply changes at the same granularity level, just controlled via the same common generic `ChangeProjectInfo` enum flag like we already discussed

if you want to have more project info changes applied via one single method it's possible but that might be even more confusing since certain things on `ProjectInfo` would be applied via its own pre-existing methods (i.e. `ApplyParseOptionsChanged`, `ApplyCompilationOptionsChanged` and so on) but others would be "grouped" together as suggested here.

Reply:
also, that would require to create a new DTO - one that captures only the "grouped" properties that change, not all of the ones in `ProjectInfo`